### PR TITLE
Handle Registering for Multiple Semesters (Summer Shenanigans)

### LINKED ISSUE #820: Handle Summer Course Registration
Our redirect to the course onboarding flow is based on if a professor's last registered semester is different than the semester of the courses we receive from Khoury admin. 

However the definition of that lastRegisteredSemester is a little cloudy if a professor is teaching both summer full and summer 1 courses. If they try to register for both, what lastRegisteredSemester should we use? What semester value do we use for deciding if there are any pending courses for registration? What happens if the courses in the Khoury Data Params have different semesters upon login?

Look into this problem and create tickets for whatever solution you come up with. Implement those changes in a PR.

This will require changes in:
* deciding if there are any pending courses, in ProfileService
* setting the lastRegisteredSemester in the course registration endpt

### PR DESCRIPTION
**Pull request by brianyyang**
# Description

Refactors our pending courses parsing to consider Summer 1 and Summer Full to be the same semester.

Previously, our redirect to the course onboarding flow is based on if a professor's last registered semester is different than the semester of the courses we receive from Khoury admin. This raises the issue of Summer 1/Summer Fall registration as if you register for both at the same time, it was unclear which to set `lastRegisteredSemester` to and how to handle that when later registering for Summer 2 classes.

Closes #820 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This has not been manually tested but we wrote two unit tests in `profile.service.spec.ts` for the case in which we want Summer 2 courses to be pushed to pending courses and `lastRegisteredSemester` is either Summer 1 or Summer 2. All the previous unit tests pass still as well so we probably didn't break anything.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/server/src/profile/profile.service.ts [hunk 1]
```diff
@@ -23,27 +26,47 @@ export class ProfileService {
...
+
+  // hacky way to check if two semester codes from Khoury have the same registration window
+  private isSameRegistrationTime(semester1: string, semester2: string) {
+    const year1 = Number(semester1.slice(0, 4));
+    const year2 = Number(semester2.slice(0, 4));
+    const semesterCode1 = semester1.slice(-2);
+    const semesterCode2 = semester2.slice(-2);
+    // we want to treat Summer 1 and Summer Full as the same window
+    // for profs to register classes for both at the same time
+    if (
```
Comment (dfarooq610) [inline packages/server/src/profile/profile.service.ts:62]: Nit: just save the Summer1 and Summer full codes as constants so you don't have to access them twice

**Review by dfarooq610 — APPROVED** [review]
Seems pretty par for the course! Good job!

**Review by tiingweii-shii — APPROVED** [review]
Looks good :D

File: packages/server/src/profile/profile.service.ts (hunk 1 above)
Comment (brianyyang) [reply packages/server/src/profile/profile.service.ts:64]: yes

**Pull request merged**